components/credentials/src/main/kotlin/org/cloudfoundry/credhub/credential/CertificateCredentialValue.kt

```diff
@@ -66,7 +69,8 @@ class CertificateCredentialValue : CredentialValue {
         certificateAuthority: Boolean,
         selfSigned: Boolean,
         generated: Boolean?,
```

Comment:
I understand that this value is nullable. I dislike that very much. I very much would prefer this field to be a boolean. 

Developers often use boolean + null when they need a type with 3 values, I think that this is a mistake. We should created a type that clearly represents the 3 values that we want.

Reply:
You may have meant to make this comment on a different line. The `generated` field existed before our PR. For historical purposes, I believe it was nullable because there was no way to know what the `true` or `false` value should be when it was originally added to the CredHub database. Perhaps an enum would have been better.

Regardless, we've updated the new `durationOverridden` field we've introduced to not be nullable. We did not touch the existing `generated` field.

Reply:
I did mean to comment on this line, and I did know that you did not write it :).

I meant to say that I realize that other fields are currently nullable, and I don't like that :)